### PR TITLE
Compatibility update for v3.0.0

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Ensures compatibility to precice:develop.

### CODE REVIEW

File: fenicsprecice/fenicsprecice.py [hunk 1]
```diff
@@ -251,9 +249,6 @@ def write_data(self, write_function):
         assert (self._write_function_type == determine_function_type(w_func, self._fenics_dims))
         assert (write_function.function_space() == self._write_function_space)
```
Comment (BenjaminRodenberg) [inline fenicsprecice/fenicsprecice.py:250]: Here, we have `assert (...)` (with blank), not `assert(...)` (without blank)

**Review by BenjaminRodenberg — COMMENTED** [review]
Since the test with the partitioned heat tutorial is failing: I guess we also need to update the tutorials. Into the rabbit hole :rabbit2:

**Comment by BenjaminRodenberg** [comment]
I also added the `v2.0.0` milestone here, because merging this PR would be breaking due to the changes in the action interface. We could also design this in a non-breaking fashion and hide the breaking changes of preCICE `v3` inside of the adapter, but this might be more confusing than useful, I guess.

File: fenicsprecice/fenicsprecice.py (hunk 1 above)
Comment (IshaanDesai) [reply fenicsprecice/fenicsprecice.py:250]: Every `assert` is now of the style `assert (...)`.

Comment (BenjaminRodenberg) [reply fenicsprecice/fenicsprecice.py:250]: All fine. As long as it is consistent :)

File: fenicsprecice/expression_core.py [hunk 2]
```diff
@@ -128,10 +128,10 @@ def is_scalar_valued(self):
         """
         try:
             if self._vals.ndim == 1:
-                assert(self._function_type is FunctionType.SCALAR)
+                assert (self._function_type is FunctionType.SCALAR)
```
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply fenicsprecice/expression_core.py:131]: I can live with both, as long as it is consistently enforced :)

**Review by IshaanDesai — APPROVED** [review]

**Comment by BenjaminRodenberg** [comment]
Since this PR is already quite old: At the moment it is not on the first place of my list of priorities, but I expect to work with preCICE v3 + FEniCS within the next few months again. We can wait with this issue until then and I will occasionally review and merge. If something else pops up in the meantime, we can, of course, take care of this issue earlier.

**Comment by IshaanDesai** [comment]
> Since this PR is already quite old: At the moment it is not on the first place of my list of priorities, but I expect to work with preCICE v3 + FEniCS within the next few months again. We can wait with this issue until then and I will occasionally review and merge. If something else pops up in the meantime, we can, of course, take care of this issue earlier.

I plan to work on this towards the end of this month. Most of the updates in the adapter itself have been done. The major work is actually in updating the tutorials in https://github.com/precice/tutorials/pull/333.

**Comment by BenjaminRodenberg** [comment]
@IshaanDesai I'll try to wrap this PR up soon. Main problem here are the unit tests. I assume the tutorial tests should work as soon as https://github.com/precice/tutorials/pull/333 is merged.

File: fenicsprecice/adapter_core.py [hunk 3]
```diff
@@ -415,31 +412,36 @@ def get_coupling_triangles(function_space, coupling_subdomain, precice_edge_dict
...
             e1, e2, e3 = list(edges(cell))
-            if all(edge in precice_edge_dict.keys() for edge in [e1.index(), e2.index(), e3.index()]):
-                edges_ids.append([e1.index(), e2.index(), e3.index()])
-
-    return np.array(edges_ids)
+            if all(edge_ids in fenics_edge_ids for edge_ids in [e1.index(), e2.index(), e3.index()]):
+                v1, v2 = vertices(e1)
+                _, v3 = vertices(e2)
+                assert (v3 != v1)
+                assert (v2 != v2)
```
Comment (BenjaminRodenberg) [inline fenicsprecice/adapter_core.py:439]: I found this problem by running `channel-transport-reaction`. But even fixing this obviously wrong assertion does not help. There seems to be some issue with the function `get_coupling_triangles`. We obviously need a test for this function and need to fix it. @IshaanDesai can you take a look?

**Pull request merged**